provider/kubernetes/kubernetes.go

```diff
@@ -421,9 +435,45 @@ func getRuleForPath(pa extensionsv1beta1.HTTPIngressPath, i *extensionsv1beta1.I
 		}
 		rules = append(rules, ruleTypeReplacePath+":"+rootPath)
 	}
+
+	if requestModifier := getStringValue(i.Annotations, annotationKubernetesRequestModifier, ""); requestModifier != "" {
+		rule, err := parseRequestModifier(requestModifier, ruleType)
+		if err != nil {
+			return "", err
+		}
+
+		rules = append(rules, rule)
+	}
 	return strings.Join(rules, ";"), nil
 }
 
+func parseRequestModifier(requestModifier, ruleType string) (string, error) {
+	if strings.Trim(requestModifier, " :") == "" {
+		return "", fmt.Errorf("could not get rule due to empty rule: %q", requestModifier)
```

Comment:
I'd simplify to `"rule %q is empty"` as the [calling site](https://github.com/dtomcej/traefik/blob/844d02f11372b98dfb1943e312f37675dd81fee6/provider/kubernetes/kubernetes.go#L260) already provides context that the error is about rule retrieval.